mllib/src/main/scala/org/apache/spark/ml/source/image/ImageFileFormat.scala

```diff
@@ -98,3 +103,163 @@ private[image] class ImageFileFormat extends FileFormat with DataSourceRegister
     }
   }
 }
+
+object ImageFileFormat {
```

Comment:
private[spark]?

Reply:
Similar to `BinaryFileFormat`, sometimes user may want to get schema by `ImageFileFormat.schema`, so that keep it public is useful.

Reply:
`BinaryFileFormat` has to be private. We only don't do `private[sql]` or `private[spark]` in execution and catalyst modules because we explicitly mention that those modules are private as of SPARK-16813 and SPARK-16964

I don't think we should keep non-API instances as public.